### PR TITLE
[Backport 7.51.x] Ensure host metadata information information is sent first

### PR DESCRIPTION
**Pull request by GustavoCaso**
Backport https://github.com/DataDog/datadog-agent/pull/23620

### CODE REVIEW

File: comp/metadata/runner/runnerimpl/runner.go [hunk 1]
```diff
@@ -137,12 +158,25 @@ func (r *runnerImpl) handleProvider(p MetadataProvider) {
...
-	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers))
-	for _, optionaP := range r.providers {
-		if p, isSet := optionaP.Get(); isSet {
-			go r.handleProvider(p)
+	r.log.Debugf("Starting metadata runner with %d priority providers and %d regular providers", len(r.priorityProviders), len(r.providers))
+
+	go func() {
+		for _, optionaP := range r.priorityProviders {
+			if p, isSet := optionaP.Get(); isSet {
+				// Execute syncronously the priority provider
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply comp/metadata/runner/runnerimpl/runner.go:166]: I don't think we need since the rest of the code doesn't in this version.

**Review by hush-hush — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

File: comp/metadata/runner/runnerimpl/runner.go [hunk 2]
```diff
@@ -27,13 +27,18 @@ func Module() fxutil.Module {
 // MetadataProvider is the provider for metadata
 type MetadataProvider func(context.Context) time.Duration
 
+// PriorityMetadataProvider is the provider for metadata that needs to be execute at start time of the agent.
+// Right now the agent needs the host metadata provider to execute first to ensure host tags are being reported correctly.
+// This is a temporary workaorund until we figure a more permanent solution in the backend
+// If you need to use this provide please contact the agent-shared-components team
```
Comment (urseberry) [inline comp/metadata/runner/runnerimpl/runner.go:33]:
```suggestion
// PriorityMetadataProvider is the provider for metadata that needs to be executed at start time of the Agent.
// Right now, the Agent needs the host metadata provider to execute first to ensure host tags are being reported correctly.
// This is a temporary workaround until we figure out a more permanent solution in the backend
// If you need to use this provider, please contact the agent-shared-components team
```

**Review by urseberry — APPROVED** [review]
Left some non-blocking feedback.

**Pull request merged**